samples/NativeAOT/MainApplication.cs

```diff
@@ -0,0 +1,22 @@
+using Android.Runtime;
+using Android.Util;
+
+/// <summary>
+/// NOTE: This class is not required, but used for testing Android.App.Application subclasses.
+/// </summary>
+[Register ("my/MainApplication")] // Required for typemap in NativeAotTypeManager
```

Comment:
Instead of using `[Register]`, you should be able to use `ApplicationAttribute.Name`:

```csharp
[Application (Name="my/MainApplication")]
partial class MainApplication {
}
```

This should also be true for `MainActivity` and using `ActivityAttribute.Name`.